node/pkg/governor/governor_test.go

```diff
@@ -0,0 +1,694 @@
+	assert.Equal(t, uint64(0), valuePending)
+}
+
+func TestVaaForUninterestingPayloadType(t *testing.T) {
+	ctx := context.Background()
+	gov, err := newChainGovernorForTest(ctx)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, gov)
+
+	emitterAddr, _ := vaa.StringToAddress("0x0290fb167208af455bb137780163b7b7a9a10c16")
+	var payload = []byte{2, 97, 97, 97, 97, 97}
+
+	msg := common.MessagePublication{
+		TxHash:           hashFromString("0x06f541f5ecfc43407c31587aa6ac3a689e8960f36dc23c332db5510dfc6a4063"),
+		Timestamp:        time.Unix(int64(1654543099), 0),
+		Nonce:            uint32(1),
+		Sequence:         uint64(1),
+		EmitterChain:     vaa.ChainIDEthereum,
+		EmitterAddress:   emitterAddr,
+		ConsistencyLevel: uint8(32),
+		Payload:          payload,
+	}
+
+	canPost, err := gov.ProcessMsgForTime(&msg, time.Now())
+	assert.NoError(t, err)
+
+	numTrans, valueTrans, numPending, valuePending := gov.GetStatsForAllChains()
+	assert.Equal(t, true, canPost)
+	assert.Equal(t, 0, numTrans)
+	assert.Equal(t, uint64(0), valueTrans)
+	assert.Equal(t, 0, numPending)
+	assert.Equal(t, uint64(0), valuePending)
+}
+
+// Note this method assumes 18 decimals for the amount.
+func buildMockTransferPayloadBytes(
+	t uint8,
+	tokenChainID vaa.ChainID,
+	tokenAddrStr string,
+	toChainID vaa.ChainID,
+	toAddrStr string,
+	amtFloat float64,
+) []byte {
+	bytes := make([]byte, 101)
+	bytes[0] = t
+
+	amtBigFloat := big.NewFloat(amtFloat)
+	amtBigFloat = amtBigFloat.Mul(amtBigFloat, big.NewFloat(100000000))
+	amount, _ := amtBigFloat.Int(nil)
+	amtBytes := amount.Bytes()
+	if len(amtBytes) > 32 {
+		panic("amount will not fit in 32 bytes!")
+	}
+	copy(bytes[33-len(amtBytes):33], amtBytes)
+
+	tokenAddr, _ := vaa.StringToAddress(tokenAddrStr)
+	copy(bytes[33:65], tokenAddr.Bytes())
+	binary.BigEndian.PutUint16(bytes[65:67], uint16(tokenChainID))
+	toAddr, _ := vaa.StringToAddress(toAddrStr)
+	copy(bytes[67:99], toAddr.Bytes())
+	binary.BigEndian.PutUint16(bytes[99:101], uint16(toChainID))
+	// fmt.Printf("Bytes: [%v]", hex.EncodeToString(bytes))
+	return bytes
+}
+
+func TestBuidMockTransferPayload(t *testing.T) {
+	tokenAddrStr := "0xDDb64fE46a91D46ee29420539FC25FD07c5FEa3E" //nolint:gosec
+	toAddrStr := "0x707f9118e33a9b8998bea41dd0d46f38bb963fc8"
+	payloadBytes := buildMockTransferPayloadBytes(1,
+		vaa.ChainIDEthereum,
+		tokenAddrStr,
+		vaa.ChainIDPolygon,
+		toAddrStr,
+		1.25,
+	)
+
+	payload, err := vaa.DecodeTransferPayloadHdr(payloadBytes)
+
+	expectedTokenAddr, _ := vaa.StringToAddress(tokenAddrStr)
+	expectedToAddr, _ := vaa.StringToAddress(toAddrStr)
+
+	expectedAmount := big.NewInt(125000000)
+
+	assert.NoError(t, err)
+	assert.Equal(t, uint8(1), payload.Type)
+	assert.Equal(t, vaa.ChainIDEthereum, payload.OriginChain)
+	assert.Equal(t, expectedTokenAddr, payload.OriginAddress)
+	assert.Equal(t, vaa.ChainIDPolygon, payload.TargetChain)
+	assert.Equal(t, expectedToAddr, payload.TargetAddress)
+	assert.Equal(t, 0, expectedAmount.Cmp(payload.Amount))
```

Comment:
It would be nicer to just create an `expected` payload and then make a single assertion against it:

```
assert.Equal(t, expected, payload)
```